temporian/core/operators/add_index.py

```diff
@@ -143,22 +143,76 @@ def _normalize_index_to_set(
 def add_index(input: Node, index_to_add: Union[str, List[str]]) -> Node:
     """Adds one or more features as index in a node.
 
-    Examples:
-        Given an input `Node` with index names ['A', 'B', 'C'] and features
-        names ['X', 'Y', 'Z']:
-
-        3. `add_index(input, feature_names=['X', 'Y'])`
-           Output `Node` will have index names ['A', 'B', 'C', 'X', 'Y'] and
-           features names ['Z'].
+    Usage example:
+        ```python
+        >>> a_evset = tp.event_set(
+        ...     timestamps=[1, 2, 1, 0, 1, 1],
+        ...     features={
+        ...         "store": [1, 1, 1, 2, 2, 2],
```

Comment:
Ideally, all the example would use the same feature/indexes.
I am not sure using the store/product/sales actually helps understanding the example (unless we really use it everywhere). I would find it easier if features were just "f1, f2, ..." or "a", "b", "c". Ideally, all our examples would use the same implicit type of example.

What about something like:

evset = event_set(
    timestamps=[0.1, 0.2, 0.3, 0.4],
    features={
        "a": ["x", "x", "y", "y"],
        "b": [1, 1, 2, 3],
        "c": ["A", "B", "C", "D"],
        "d": [10, 11, 12, 13],
    }
)

tp.add_index(evset.node(), ["a", "b"])

or

evset = event_set(
    timestamps=[0.1, 0.2, 0.3, 0.4],
    features={
        "cat_1": ["x", "x", "y", "y"],
        "cat_2": ["A", "B", "C", "D"],
        "num_1": [1, 1, 2, 3],
        "num_2": [10, 11, 12, 13],
    }
)